sdk/dotnet/Pulumi/Deployment/Deployment.cs

```diff
@@ -157,12 +160,23 @@ internal Deployment(IEngine engine, IMonitor monitor, TestOptions? options)
             _isDryRun = options?.IsPreview ?? true;
             _stackName = options?.StackName ?? "stack";
             _projectName = options?.ProjectName ?? "project";
+            _organizationName = null;
             this.Engine = engine;
             this.Monitor = monitor;
             _runner = new Runner(this, deploymentLogger);
             _logger = new EngineLogger(this, deploymentLogger, this.Engine);
         }
 
+        string IDeployment.OrganizationName {
+            get
+            {
+                if (string.IsNullOrEmpty(_organizationName))
+                {
+                    throw new Exception("organization is not available");
```

Comment:
Should this include what users can do when they are unit testing but using `OrganisationName`? It will throw but doesn't tell anything actionable. Maybe:
```
organization name is not available, either you are running an old version of Pulumi which doesn't pass the organization information to the program or running the program inside tests that don't have an organization. Set PULUMI_ORGANIZATION environment variable from which the organisation name is retrieved. 
```

Reply:
Could do, could update the other nodejs and python errors similarly.